docs/user_guide/backend_configuration.rst

```diff
@@ -127,6 +127,20 @@ We can confirm these values are saved by re-printing that particular dataset con
       compression method : Zstd
       compression options : {'clevel': 3}
 
+Then we can use this configuration to write the NWB file:
+
+.. code-block:: python
+
+    from neuroconv.configure_backend import configure_backend, BACKEND_NWB_IO
+
+    dataset_configurations["acquisition/MyTimeSeries/data"] = dataset_configuration
+
+    configure_backend(nwbfile=nwbfile, backend_configurations=backend_configurations)
+    IO = BACKEND_NWB_IO[backend_configuration.backend]
+
+    with IO("my_nwbfile.nwb", mode="w") as io:
+        io.write(nwbfile)
```

Comment:
I think it might make sense to have a convenience function that bundles this as a function, to avoid the mistake of a user configuration a dataset with one backend and then accidentally using a different IO

Reply:
The current 'convenience' function is the `make_or_load_nwbfile` context as used in the generic tools section, which will write the file using the correct IO

But yes, @h-mayorquin has proposed breaking up that magical context into distinct pieces, one of which would be `write_nwbfile` (which I guess would no longer be a context...)

Things for a follow-up. Just trying to get this basic documentation in as a start